### PR TITLE
Added Knights Tour Problem

### LINKED ISSUE #7100: Knight Tour Problem in C++
### Description

Documentation of Knight Tour Problem in C++

### Domain

C/CPP

### Type of Contribution

Documentation

### Code of Conduct

- [X] I follow [Contributing Guidelines](https://github.com/girlscript/winter-of-contributing/blob/main/.github/CONTRIBUTING.md) & [Code of conduct](https://github.com/girlscript/winter-of-contributing/blob/main/.github/CODE_OF_CONDUCT.md) of this project.

### PR DESCRIPTION
**Pull request by aman12207**
<hr>

## Description 📜

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. 
Fixes #7100

<hr>

## Type of change 📝

- [x] Documentation (Content Creation in the form of codes or tutorials)

<hr>

## Domain of Contribution 📊

- [x] C/CPP
<hr>

## Checklist ✅

- [x] I follow [Contributing Guidelines](https://github.com/girlscript/winter-of-contributing/blob/main/.github/CONTRIBUTING.md) & [Code of conduct](https://github.com/girlscript/winter-of-contributing/blob/main/.github/CODE_OF_CONDUCT.md) of this project.
- [x] I have performed a self-review of my own code or work.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] My changes generates no new warnings.
- [x] I'm GWOC'21 contributor

<hr>

## Screenshots / Gif (Optional) 📸

<hr>

### CODE REVIEW

File: C_CPP/Algorithmic Approaches/Backtracking/Knights_Tour_problem.md [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+    knightTour(chess, r + 2, c - 1, move + 1);          //RECURSIVE CALL FOR MOVING TWO STEPS UPWARD(2↑) 1 STEP LEFT(1←)
+    knightTour(chess, r + 1, c - 2, move + 1);          //RECURSIVE CALL FOR MOVING TWO STEPS UPWARD(1↑) 1 STEP LEFT(2←)
+    knightTour(chess, r - 1, c - 2, move + 1);          //RECURSIVE CALL FOR MOVING TWO STEPS DOWNWARD(1↓) 1 STEP LEFT(2←)
+    knightTour(chess, r - 2, c - 1, move + 1);          //RECURSIVE CALL FOR MOVING TWO STEPS DOWNWARD(2↓) 1 STEP LEFT(1←)
+    chess[r][c] = 0;
+}
+
+int main()
+{
+    int n, r, c;
```
Comment (apeksha235) [inline C_CPP/Algorithmic Approaches/Backtracking/Knights_Tour_problem.md:58]: add input and output code snippet as well

**Review by apeksha235 — CHANGES_REQUESTED** [review]
@aman12207 Good work! just make this change

**Review by apeksha235 — APPROVED** [review]
@Suvraneel since the output is too long, it can't be added to the file. Looks good to me besides that, please merge it

**Comment by Suvraneel** [comment]
@aman12207  Noice 👍🏻  LGTM ❤️

**Comment by aman12207** [comment]
THANKS...

**Pull request merged**